File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        curr = 0
        maximum = 0
        for n in nums:
            if n != 1:
                curr = 0
            else:
                curr += 1
                maximum = max(curr, maximum)
                
        return maximum
        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        const ans = [];
        for (let i = 0; i < nums.length; i++) {
            let pre = i === 0 ? 1 : nums.slice(0,i)
                .reduce((acc, curr) => acc * curr, 1);
            let post = i === nums.lenght - 1 ? 1 : nums.slice(i+1)
                .reduce((acc, curr) => acc * curr, 1);

            ans.push(pre*post);
        }
        return ans;
    }
}
